packages/core-mobile/app/services/bridge/UnifiedBridgeService.ts

```diff
@@ -133,22 +82,23 @@ export class UnifiedBridgeService {
     sourceNetwork: Network
     targetNetwork: Network
   }): Promise<bigint> {
-    const feeMap = await this.service.getFees({
-      asset,
-      amount,
-      targetChain: await this.buildChain(targetNetwork),
-      sourceChain: await this.buildChain(sourceNetwork)
-    })
+    const feeMap = lowerCaseKeys(
+      await this.service.getFees({
+        asset,
+        amount,
+        targetChain: await this.buildChain(targetNetwork),
+        sourceChain: await this.buildChain(sourceNetwork)
+      })
+    )
+
+    const identifier =
+      asset.type === TokenType.NATIVE ? asset.symbol : asset.address
 
-    if (isNativeAsset(asset)) {
-      // todo: handle this properly
-      return 0n
-    } else if (isErc20Asset(asset) && asset.address) {
-      const address = asset.address.toLowerCase() as `0x${string}`
-      return feeMap[address] ?? 0n
-    } else {
+    if (!identifier) {
       throw new Error('invalid asset')
     }
+
+    return feeMap[identifier.toLowerCase()] ?? 0n
```

Comment:
we probably should do undefined here instead of 0n. that way we don't ever display 0 in the UI.

Reply:
fixed.